### ISSUE TITLE
Exceptions for better performance

### ISSUE THREAD

**Issue by lefticus** [lefticus/cpp_weekly#158]
A comparison of the monadic call chain interface from std::optional in C++23 and using exceptions to handle error cases instead.

**Comment by usefulcat** [comment]
I would upvote this 10x if I could. I have long theorized, but never had the chance/taken the time to prove, that using exceptions to report uncommon conditions can be measurably faster than manually checking and propagating error return values, especially with a deep-ish call stack.

**Comment by fcolecumberri** [comment]
could you add `std::expected` to the comparison?

**Comment by lefticus** [comment]
> could you add `std::expected` to the comparison?

Good idea. I've observed that it's shockingly easy to use std::expected inefficiently.

*cough* implicit conversions *cough*

**Comment by lefticus** [comment]
Let's just reference all of the work by @kammce and consider this closed.

https://www.youtube.com/watch?v=wNPfs8aQ4oo
